cmd/fleet/serve.go

```diff
@@ -614,6 +613,33 @@ the way that the Fleet server works.
 				initFatal(errors.New("Error generating random instance identifier"), "")
 			}
 
+			// Perform a cleanup of cron_stats outside of the cronSchedules because the
+			// schedule package uses cron_stats entries to decide whether a schedule will
+			// run or not (see https://github.com/fleetdm/fleet/issues/9486).
+			go func() {
+				cleanupCronStats := func() {
+					level.Debug(logger).Log("msg", "cleaning up cron_stats")
+					// Datastore.CleanupCronStats should be safe to run by multiple fleet
+					// instances at the same time and it should not be an expensive operation.
+					if err := ds.CleanupCronStats(ctx); err != nil {
+						level.Info(logger).Log("msg", "failed to clean up cron_stats", "err", err)
+					}
+				}
+
+				cleanupCronStats()
+
+				cleanUpCronStatsTick := time.NewTicker(1 * time.Hour)
```

Comment:
I'm guessing we run this every hour instead of every two hours, just to be safe?

Reply:
I believe that we run this every hour to decrease the wait until the entries are cleaned up.

@gillespi314